extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnectDirectory.java

```diff
@@ -118,5 +136,54 @@ public String create(GuacamoleConfiguration config) throws GuacamoleException {
 
         return newConnectionId;
     }
+    
+    /**
+     * Checks the provided GuacamoleConfiguration to make sure the parameters
+     * in the configuration are allowed by the configuration.  If the
+     * allowed parameter configuration option is set, all parameters not
+     * specified will be removed from the config.  If the denied parameter
+     * configuration option is set, any parameters specified by that will be
+     * removed from the configuration.  If neither are set then all parameters
+     * are allowed.
+     * 
+     * @param config
+     *     The GuacamoleConfiguration containing parameters to check against
+     *     configured allowed and denied values.
+     * 
+     * @throws GuacamoleException 
+     *     If guacamole.properties cannot be parsed.
+     */
+    private void checkConfigParameters(GuacamoleConfiguration config) throws GuacamoleException {
+        
+        // Get configuration items.
+        List<String> allowedParams = confService.getAllowedParameters();
+        List<String> deniedParams = confService.getDeniedParameters();
+        
+        // Loop through parameters and remove them if they are not allowed.
+        Set<String> setParams = new HashSet<>(config.getParameterNames());
+        for (String key : setParams) {
+            
+            if (allowedParams != null
+                    && !allowedParams.isEmpty()
+                    && !allowedParams.contains(key)) {
+                
+                logger.debug("Parameter \"{}\" is not allowed and will be removed.",
+                        key);
+                config.unsetParameter(key);
+            
+            }
+            
+            if (deniedParams != null
+                    && !deniedParams.isEmpty()
+                    && deniedParams.contains(key)) {
+                
+                logger.debug("Parameter \"{}\" has been denied and will be removed.",
+                        key);
+                config.unsetParameter(key);
+                
+            }
+        }
```

Comment:
`config.getParameters().keySet().retainAll()` and `config.getParameters().keySet().removeAll()` would be easier than manually looping through the map.

Reply:
Wow. Yes.